core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -841,6 +856,121 @@ pathSpecification
     : pathElement (',' pathElement)*
     ;
 
+functionSpecification
+    : FUNCTION qualifiedName parameterDeclarationList
+        returnsClause routineCharacteristic*
+        (returnStatement | compoundStatement)
+    ;
+
+parameterDeclarationList
+    : '(' (parameterDeclaration (',' parameterDeclaration)*)? ')'
+    ;
+
+parameterDeclaration
+    : identifier? type
+    ;
+
+returnsClause
+    : RETURNS type
+    ;
+
+routineCharacteristic
+    : LANGUAGE identifier               #languageCharacteristic
+    | NOT? DETERMINISTIC                #deterministicCharacteristic
+    | RETURNS NULL ON NULL INPUT        #returnsNullOnNullInputCharacteristic
+    | CALLED ON NULL INPUT              #calledOnNullInputCharacteristic
+    | SECURITY (DEFINER | INVOKER)      #securityCharacteristic
+    | COMMENT string                    #commentCharacteristic
+    ;
+
+routineStatement
+    : returnStatement
+    | assignmentStatement
+    | simpleCaseStatement
+    | searchedCaseStatement
+    | ifStatement
+    | iterateStatement
+    | leaveStatement
+    | compoundStatement
+    | loopStatement
+    | whileStatement
+    | repeatStatement
+    ;
+
+returnStatement
+    : RETURN valueExpression
+    ;
+
+assignmentStatement
+    : SET identifier EQ expression
+    ;
+
+simpleCaseStatement
+    : CASE expression caseStatementWhenClause+ elseClause? END CASE
+    ;
+
+searchedCaseStatement
+    : CASE caseStatementWhenClause+ elseClause? END CASE
+    ;
+
+caseStatementWhenClause
+    : WHEN expression THEN sqlStatementList
+    ;
```

Comment:
For simple case, this should allow multiple operands.

Reply:
We don't support this for CASE expressions.